src/me/truemb/rentit/utils/chests/ChestsUtils.java

```diff
@@ -0,0 +1,159 @@
+package me.truemb.rentit.utils.chests;
+
+import com.sk89q.worldedit.math.BlockVector3;
+import me.truemb.rentit.enums.RentTypes;
+import me.truemb.rentit.main.Main;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ChestsUtils {
+    private Main instance;
+
+    public ChestsUtils(Main plugin) {
+        this.instance = plugin;
+    }
+
+    /**
+     * Returns a list of wrappers manipulate any chest supported and enabled.
+     * @param shopId The id of the shop to get all the chests from.
+     */
+    public List<SupportedChest> getShopChests(int shopId) {
```

Comment:
`getShopChests` now returns a list of `SupportedChest` which is a wrapper for any chest that we find inside the shop. Implementations for basic operations like querying the list of items or adding new ones might change between each type of supported chest. Also, new types could be added like vanilla BARREL or other third party plugins.